lib/cadet/assessments/assessments.ex

```diff
@@ -72,36 +59,35 @@ defmodule Cadet.Assessments do
   end
 
   def publish_assessment(id) do
-    id
-    |> get_assessment()
-    |> change(%{is_published: true})
-    |> Repo.update()
-  end
-
-  def get_question(id) do
-    Repo.get(Question, id)
-  end
-
-  def get_assessment(id) do
-    Repo.get(Assessment, id)
+    simple_update(
+      Assessment,
+      id,
+      using: &Assessment.changeset/2,
+      params: %{is_published: true}
+    )
   end
 
   def create_question_for_assessment(params, assessment_id)
       when is_binary(assessment_id) or is_number(assessment_id) do
-    assessment = get_assessment(assessment_id)
-    create_question_for_assessment(params, assessment)
-  end
-
-  def create_question_for_assessment(params, assessment) do
     Repo.transaction(fn ->
-      assessment = Repo.preload(assessment, :questions)
-      questions = assessment.questions
+      assessment =
+        Assessment
+        |> where([assessment], assessment.id == ^assessment_id)
+        |> join(
```

Comment:
Preferable to use one letter bindings, i.e.
`|> join(:left, [a], q in Question, q.assessment_id == a.id)`